### PR TITLE
fix(request-manager): localhost connection without proxy

### PR DESCRIPTION
**Pull request by karliatto**
## Description
Fix request manager proxy to allow all localhost connections without proxy.

### CODE REVIEW

**Review by mroz22 — APPROVED** [review]

File: packages/request-manager/src/interceptor.ts [hunk 1]
```diff
@@ -130,9 +132,9 @@ const interceptHttps = (interceptorOptions: InterceptorOptions) => {
                 delete overloadedOptions.headers['Proxy-Authorization'];
             }
             // Requests to localhost should not use the proxy.
-            const shouldIntercept = overloadedOptionsUrl.hostname !== '127.0.0.1';
+            const isNotLocalhost = !isLocalhost(overloadedOptionsUrl.hostname);
```
Comment (szymonlesisz) [inline packages/request-manager/src/interceptor.ts:135]: shouldnt we have this block of code, and the block above (http.request) done as one function? its like 30+ duplicated lines

Comment (mroz22) [reply packages/request-manager/src/interceptor.ts:135]: thats right, `isNotLocalhost` util would make sense here

Comment (karliatto) [reply packages/request-manager/src/interceptor.ts:135]: Yes, I created issue https://github.com/trezor/trezor-suite/issues/6674 to address that.

Comment (mroz22) [reply packages/request-manager/src/interceptor.ts:135]: ok, merging.

**Pull request merged**